risc0/zkvm/src/host/server/exec/executor.rs

```diff
@@ -257,24 +287,43 @@ impl<'a> ExecutorImpl<'a> {
     where
         F: FnMut(Segment) -> Result<Box<dyn SegmentRef>>,
     {
-        let mut guest_session = match self.run_guest_only_with_callback(callback) {
-            Ok(session) => session,
-            Err(e) => return Err(ExecutorError::Error(e)),
-        };
-        match guest_session.exit_code {
+        let start = std::time::Instant::now();
+        let mut guest_session = self
+            .run_guest_only_with_callback(callback)
+            .map_err(ExecutorError::from)?;
+        let result = match guest_session.exit_code {
             ExitCode::Fault => {
-                let fault_checker_session = match self.run_fault_checker() {
-                    Ok(session) => session,
-                    Err(e) => return Err(ExecutorError::Error(e)),
-                };
-                for segment in fault_checker_session.segments {
-                    guest_session.segments.push(segment);
-                }
-                guest_session.journal = fault_checker_session.journal;
-                Err(ExecutorError::Fault(guest_session))
+                let Session {
+                    segments, journal, ..
+                } = self.run_fault_checker().map_err(ExecutorError::from)?;
+                segments
+                    .into_iter()
+                    .for_each(|s| guest_session.segments.push(s));
+                guest_session.journal = journal;
+                Err(guest_session.into())
             }
             _ => Ok(guest_session),
+        };
+
+        if log::log_enabled!(target: "RISC0_EXECUTION_STATS", log::Level::Info) {
```

Comment:
Any reason to use a dedicated var vs `RUST_LOG="executor=info"` syntax?

Reply:
It's not a dedicated var, it's actually a string value we should pass. similar to `RUST_LOG="executor=info"`. I chose a longer value though as originally, when dealing with `tracing` filters we were matching module names as well (and executor was one of them). 

Moving to `env_logger` we can use `executor` as a shorter filter though :)